crucible-llvm/src/Lang/Crucible/LLVM/SimpleLoopFixpoint.hs

```diff
@@ -96,53 +114,70 @@ data FixpointEntry sym tp = FixpointEntry
   }
 
 instance OrdF (W4.SymExpr sym) => OrdF (FixpointEntry sym) where
-  compareF x y = case compareF (headerValue x) (headerValue y) of
-    LTF -> LTF
-    EQF -> compareF (bodyValue x) (bodyValue y)
-    GTF -> GTF
-
+  compareF x y = joinOrderingF
+    (compareF (headerValue x) (headerValue y))
+    (compareF (bodyValue x) (bodyValue y))
 instance OrdF (FixpointEntry sym) => W4.TestEquality (FixpointEntry sym) where
-  testEquality x y = case compareF x y of
-    EQF -> Just Refl
-    _ -> Nothing
+  testEquality x y = orderingF_refl $ compareF x y
 
-data MemFixpointEntry sym = forall w . (1 <= w) => MemFixpointEntry
-  { memFixpointEntrySym :: sym
-  , memFixpointEntryJoinVariable :: W4.SymBV sym w
+data MemLocation sym w = MemLocation
+  { memLocationBlock :: Natural
```

Comment:
[CI complains](https://github.com/GaloisInc/crucible/actions/runs/7637764713/job/20807303023?pr=1166#step:11:1106) due to a build error on this line:

```
[42 of 54] Compiling Lang.Crucible.LLVM.SimpleLoopFixpoint ( src/Lang/Crucible/LLVM/SimpleLoopFixpoint.hs, /home/runner/work/crucible/crucible/dist-newstyle/build/x86_64-linux/ghc-8.10.7/crucible-llvm-0.5/build/Lang/Crucible/LLVM/SimpleLoopFixpoint.o, /home/runner/work/crucible/crucible/dist-newstyle/build/x86_64-linux/ghc-8.10.7/crucible-llvm-0.5/build/Lang/Crucible/LLVM/SimpleLoopFixpoint.dyn_o )

src/Lang/Crucible/LLVM/SimpleLoopFixpoint.hs:124:25: error:
Error:     Not in scope: type constructor or class ‘Natural’
    |
124 |   { memLocationBlock :: Natural
    |                         ^^^^^^^
```